parts/dcosagentcustomscript.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+set -e
+
+update_attribute()
+{
+    if [ ! -a /etc/mesosphere/roles/master ];
+    then
+        echo adding FD / UD attributes
+        meta=$( curl http://169.254.169.254/metadata/v1/InstanceInfo )
+        # parsing with bash to avoid jq install. This might break when the metadata service changes

```

Comment:
Strongly agree. This will break if they ever add new fields out of order to InstanceInfo. This is why other metadata services allow for "/v1/InstanceInfo/UD" and it will just spit out the UD field.

Anyway, this is more resilient:

```
curl --silent --fail 'http://169.254.169.254/metadata/v1/InstanceInfo' | sed 's/^.*"UD": *"\([0-9]*\)".*$/\1/'
```

That will work, even if they rearrange fields, add spaces between keys/values, etc.

Additionally, ALL curl invocations must have `--fail` on them, or else they will fail with a success exit code and your scripts won't handle it correctly.